File: 153. 寻找旋转排序数组中的最小值.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int left=0,right=nums.size()-1,mid;
        while(left<=right)
        {
            mid=(left+right)/2;
            if(nums[mid]>nums[mid+1])
            {
                mid=mid+1;
                break;
            }
            else if(nums[mid]<nums[mid-1])
                break;
            
            // mid满足nums[mid-1]<nums[mid]<nums[mid+1].即mid处于升序段中,此时需判断key位于mid的左方还是右方
            else 
            {
                if(nums[0]<nums[mid-1]) // key位于mid的右方
                    right=mid+1;
                else 
                    left=mid-1; // key位于mid的左方
            }
        }
        return nums[mid];
    }
};

File: js/153. 寻找旋转排序数组中的最小值.js
/**
 * @param {number[]} nums
 * @return {number}
 */

// 旋转数组，末尾元素一定小于等于第一个元素
var findMin = function(nums) {
    
    let left=0,right=nums.length-1
    while(left<right){
        let mid=left+Math.floor((right-left)/2)
        if(nums[mid]>nums[right])
            left=mid+1
        else 
            right=mid
    }
    return nums[left] 
};
